app/src/main/java/com/example/employeeinformation/DetailsActivity.kt

```diff
@@ -0,0 +1,75 @@
+package com.example.employeeinformation
+
+import android.content.Intent
+import android.os.Bundle
+import android.text.Editable
+import android.text.TextWatcher
+import android.widget.Button
+import android.widget.EditText
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+
+class DetailsActivity : AppCompatActivity() {
+    lateinit var button: Button
+    lateinit var name: EditText
+    lateinit var role: EditText
+    lateinit var age: EditText
+    lateinit var experience: EditText
+    override fun onCreate(savedInstanceState: Bundle?) {
```

Comment:
This method is very long and it contains multiple functionality.
We can try breaking down this into seperate methods.
For example initialising views, Creating intents.  ( setOnClickListener is doing too many things)
We can even extract TextWatcher to another class